### PR TITLE
Storage STG 83 features

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

**Comment by MushMal** [comment]
Apologies for not having the context for this change. There seem to be a few rounds of changes - some are cleanups or bringing a consistency across versions. I was wondering if we could have a high-level description of the changes and some annotations in the form of comments in the review. The diff I see primarily includes a single addition of the Filter. 

There are also a set of linter rules warnings/errors when run spectral locally.

**Comment by seanmcc-msft** [comment]
@tg-msft, could you help get traction on this?

**Comment by MushMal** [comment]
> @tg-msft, could you help get traction on this?

@seanmcc-msft, is there an issue opened for this PR where we could get more information about the change? I can't seem to find any documentation or any narrative accompanying this PR.

**Comment by seanmcc-msft** [comment]
> > @tg-msft, could you help get traction on this?
> 
> @seanmcc-msft, is there an issue opened for this PR where we could get more information about the change? I can't seem to find any documentation or any narrative accompanying this PR.

- [New blob swagger version #18275](https://github.com/Azure/azure-rest-api-specs/pull/18275) introduced the 2021-08-06 blob swagger version.  It is identical to 2021-04-08, to make the following PRs human-reviewable.
- [Added missing Premium value to AccessTier #18434](https://github.com/Azure/azure-rest-api-specs/pull/18434) fixes a swagger bug.  `Premium` has been a [valid access tier for a long time](https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types#premium-ssd).

**Review by MushMal — COMMENTED** [review]
I truly appreciate the information provided including the feature documentation allowing to navigate the change. Included a few comments.

File: specification/storage/data-plane/Azure.Storage.Files.DataLake/preview/2020-06-12/DataLakeStorage.json [hunk 1]
```diff
@@ -2585,6 +2601,25 @@
...
+      "required": false,
+      "type": "integer",
+      "x-ms-parameter-location": "client"
+    },
+    "LeaseDurationMethod": {
+      "name": "x-ms-lease-duration",
+      "x-ms-client-name": "leaseDuration",
+      "in": "header",
+      "description": "The lease duration is required to acquire a lease, and specifies the duration of the lease in seconds.  The lease duration must be between 15 and 60 seconds or -1 for infinite lease.",
+      "format": "int64",
```
[the remark this answers is not in the document]
Comment (seanmcc-msft) [reply specification/storage/data-plane/Azure.Storage.Files.DataLake/preview/2020-06-12/DataLakeStorage.json:2618]: We are using longs moving forward, this was originally int32 to historical reasons.

**Review by tg-msft — APPROVED** [review]
The new features look good to me.

**Comment by seanmcc-msft** [comment]
@tjprescott, can we merge?

**Pull request merged**